src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -182,6 +184,54 @@ def deploy_package():
                 force=force,
             )
 
+        def drop_application_before_upgrade(cascade: bool = False):
```

Comment:
had too copy this code from `NativeAppManager`, it'll be refactored into `action_drop` later